### PR TITLE
chore(upgrade): polkadot v1.10 to v1.2.0

### LINKED ISSUE #1718: Upgrade to Polkadot Release polkadot-v1.2.0
The Parity team has published new version of Polkadot: `polkadot-v1.2.0`

See https://github.com/paritytech/polkadot/releases/tag/polkadot-v1.2.0
And https://forum.polkadot.network/t/polkadot-release-analysis-v1-2-0/4451

## Upgrade Priority

See [Polkadot Release Page](https://github.com/paritytech/polkadot/releases/tag/polkadot-v1.2.0)

## Host Functions

See [Polkadot Release Page](https://github.com/paritytech/polkadot/releases/tag/polkadot-v1.2.0)

## Database Migrations

See [Polkadot Release Page](https://github.com/paritytech/polkadot/releases/tag/polkadot-v1.2.0)

## Runtime Migrations

See [Polkadot Release Page](https://github.com/paritytech/polkadot/releases/tag/polkadot-v1.2.0)

### PR DESCRIPTION
**Pull request by enddynayn**
# Goal
Update polkadot from v1.1.0 to v1.2.0.

[Polkadot release notes](https://forum.polkadot.network/t/polkadot-release-analysis-v1-2-0/4451)

Lazy Migration
https://github.com/paritytech/polkadot-sdk/pull/1363

Closes #1718

### CODE REVIEW

File: runtime/frequency/src/lib.rs [hunk 1]
```diff
@@ -688,8 +695,15 @@ impl pallet_preimage::Config for Runtime {
 		EnsureRoot<AccountId>,
 		pallet_collective::EnsureMember<AccountId, TechnicalCommitteeCollective>,
 	>;
-	type BaseDeposit = PreimageBaseDeposit;
-	type ByteDeposit = PreimageByteDeposit;
+	// type BaseDeposit = PreimageBaseDeposit;
```
Comment (enddynayn) [inline runtime/frequency/src/lib.rs:698]: rm

File: runtime/common/src/weights/pallet_preimage.rs [hunk 2]
```diff
@@ -33,6 +33,21 @@ use core::marker::PhantomData;
 /// Weights for `pallet_preimage` using the Substrate node and recommended hardware.
 pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> pallet_preimage::WeightInfo for SubstrateWeight<T> {
```
Comment (enddynayn) [inline runtime/common/src/weights/pallet_preimage.rs:35]: Running benchmarks for pallet-preimage pending.

File: e2e/proxy-pallet/proxy.test.ts [hunk 3]
```diff
@@ -31,12 +31,16 @@ describe('Proxy', function () {
     });
 
     it('Sends a transfer', async function () {
+      console.log(
```
Comment (enddynayn) [inline e2e/proxy-pallet/proxy.test.ts:34]: rm

File: runtime/frequency/src/lib.rs [hunk 4]
```diff
@@ -646,7 +649,7 @@ impl pallet_balances::Config for Runtime {
 	type WeightInfo = weights::pallet_balances::SubstrateWeight<Runtime>;
 	type MaxReserves = BalancesMaxReserves;
 	type ReserveIdentifier = [u8; 8];
-	type MaxHolds = ConstU32<0>;
+	type MaxHolds = ConstU32<1>;
```
Comment (wilwade) [inline runtime/frequency/src/lib.rs:652]: Curious why this change. I thought we didn't want to allow holds as they are going away.

**Review by wilwade — COMMENTED** [review]
One question about holds, but otherwise looks like a nice simple upgrade.

- Read through the changes

File: runtime/common/src/weights/pallet_collator_selection.rs [hunk 5]
```diff
@@ -42,10 +42,10 @@ impl<T: frame_system::Config> pallet_collator_selection::WeightInfo for Substrat
...
 		//  Measured:  `334 + b * (79 ±0)`
 		//  Estimated: `2809 + b * (2554 ±0)`
-		// Minimum execution time: 13_175_000 picoseconds.
-		Weight::from_parts(11_629_620, 2809)
-			// Standard Error: 9_686
-			.saturating_add(Weight::from_parts(3_219_955, 0).saturating_mul(b.into()))
+		// Minimum execution time: 13_026_000 picoseconds.
+		Weight::from_parts(11_528_896, 2809)
+			// Standard Error: 11_636
+			.saturating_add(Weight::from_parts(3_232_614, 0).saturating_mul(b.into()))
```
Comment (claireolmstead) [inline runtime/common/src/weights/pallet_collator_selection.rs:48]: How do you know what numbers to change all of these values to? They all seem somewhat arbitrary
.

Comment (enddynayn) [reply runtime/common/src/weights/pallet_collator_selection.rs:48]: These numbers are automatically generated when we run benchmarks against the extrinsics we create. For example, [this benchmark](https://github.com/frequency-chain/frequency/blob/main/pallets/messages/src/benchmarking.rs) tests the performance of the messages pallet. When the benchmarks are executed, they run the specified code and provide measurements of execution time and resource usage. These metrics help us understand the computational cost of transactions and determine appropriate fees.

**Review by shannonwells — APPROVED** [review]
* I reviewed Cargo.lock to check that the update SHA matches the [Polkadot release SHA](https://github.com/paritytech/polkadot-sdk/commit/63d7f90f778ae4aba0d82326cf7bba7087d96be4)
* I checked that new weights were not significantly change from before - around 2 microseconds except where noted.
* I reviewed other code changes - mostly the switch to `transferAllowDeath` and the note about MaxHolds.

**Review by mattheworris — APPROVED** [review]
- Read through the changes
- Read through the release notes and release analysis
- Executed e2e tests (All passed)

Suggestion: Consider adding https://github.com/ggwpez/substrate-scripts/blob/master/migrate-preimage-lazy.py to our `scripts`, I have found this helpful for running the scripts in the past.

**Review by wilwade — APPROVED** [review]
Ship it!

- Reviewed changes
- Ran locally
- Ran e2e tests locally
